### PR TITLE
ENV specific KMS URLs

### PR DESCRIPTION
**Pull request by jceaser**
Adding code to change the "version" parameter used when sending requests to KMS in SIT and UAT and preserving the old way for production.

### CODE REVIEW

**Comment by jceaser** [comment]
This change is needed to allow CMR to use the correct data in SIT and UAT while allowing PROD to remain on it's data set.

NOTE: this is an issue because all three environments use the same KMS server due to the way AWS and EBNET networks are configured.

**Review by eereiter — APPROVED** [review]

File: transmit-lib/src/cmr/transmit/kms.clj [hunk 1]
```diff
@@ -51,7 +84,7 @@
    :measurement-name "measurementname?format=csv"
    :concepts "idnnode?format=csv"
    :iso-topic-categories "isotopiccategory?format=csv"
-   :related-urls "rucontenttype?format=csv"
+   :related-urls (env-varient "rucontenttype?format=csv" if-not-production "&version=DRAFT")
```
Comment (sxu123) [inline transmit-lib/src/cmr/transmit/kms.clj:87]: Seems that the code doesn't need to be this fancy.  the logic is so simple
(if production?
  url
  (str url varient)) 
We don't need to pass if-not-production as a function, and have an additional env-variant function.

**Review by sxu123 — APPROVED** [review]

File: transmit-lib/src/cmr/transmit/kms.clj [hunk 2]
```diff
@@ -39,6 +39,39 @@
    :related-urls :uuid
    :granule-data-format :uuid})
 
+(defn- if-not-production
```
Comment (mitchstartzel) [inline transmit-lib/src/cmr/transmit/kms.clj:42]: nitpick: this fn would sound more clojurey if the fn name was `not-production?`

File: transmit-lib/src/cmr/transmit/kms.clj [hunk 3]
```diff
@@ -39,6 +39,39 @@
...
+      url)))
+
+(comment
+ "The following line of code is used for trasitioning CMR from SIT->UAT->PROD.
+  These three envirments all talk to production KMS due to AWS->EBNET network
+  limitations. Because we need to be able to publish data for all three envirments
+  with one host, it was decided to use the different version capabilities in KMS
+  to isolate production from the other two envirments. Specificly: rucontenttype
+  is a three level tree in SIT, but not PROD. This process is only needed durring
+  the transition and should be deleted once in production."
```
Comment (mitchstartzel) [inline transmit-lib/src/cmr/transmit/kms.clj:70]: What exactly is being deleted once in production here?

**Review by mitchstartzel — APPROVED** [review]

**Pull request merged**